File: backend/src/test/java/br/ufpr/sept/so2/shared/domain/valueobject/GrrTest.java
package br.ufpr.sept.so2.shared.domain.valueobject;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class GrrTest {

    @Test
    void aceitaGrrValido() {
        Grr grr = Grr.of("grr20241234");
        assertEquals("GRR20241234", grr.getValue());
    }

    @Test
    void rejeitaFormatoInvalido() {
        assertThrows(IllegalArgumentException.class, () -> Grr.of("20241234"));
    }
}

File: backend/src/test/kotlin/br/ufpr/sept/so2/shared/domain/valueobject/GrrTest.kt
package br.ufpr.sept.so2.shared.domain.valueobject

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class GrrTest : StringSpec({
    "aceita GRR válido" {
        Grr.of("grr20241234").value shouldBe "GRR20241234"
    }

    "rejeita formato inválido" {
        shouldThrow<IllegalArgumentException> { Grr.of("20241234") }
    }
})
